### PR TITLE
`Development`: Refactor client test routing mocks

### PR DESCRIPTION
**Pull request by MDK5**
#### General
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation, Context and Description
Previously, Artemis contained two definitions of a MockRouter, two definitions of MockActivatedRoute and about five definitions of the MockRouterLinkDirective, which were then imported to various unrealted places. This PR unifies this into one definition for each and adapts all affected imports.

Also converts two test to jest which was needed to deal with the jest mocks in the new MockRouter.

I recommend to review with whitespace changes disabled.

#### Code Review
- [x] Review 1
- [x] Review 2

### Test Coverage
unchanged

### CODE REVIEW

File: src/test/javascript/spec/component/admin/audits.component.spec.ts [hunk 1]
```diff
@@ -33,155 +34,152 @@ function getDate(isToday = true) {
...
-            mockActivatedRoute.setParameters({
-                sort: 'id,desc',
+describe('AuditsComponent', () => {
+    let comp: AuditsComponent;
+    let fixture: ComponentFixture<AuditsComponent>;
+    let service: AuditsService;
+    let mockActivatedRoute: MockActivatedRoute;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
```
Comment (lschlesinger) [inline src/test/javascript/spec/component/admin/audits.component.spec.ts:44]: you could change the order according to our guidelines here

**Review by lschlesinger — APPROVED** [review]
Nice, this makes the dependencies among tests way cleaner for the future use of `npm run test-diff`, thanks for refactoring and providing a mock that is consistently shared by all tests.

File: src/test/javascript/spec/service/navigation-util.service.spec.ts [hunk 2]
```diff
@@ -36,32 +29,37 @@ describe('Navigation Util Service', () => {
...
+        const backSpy = jest.spyOn(TestBed.inject(Location), 'back');
         // @ts-ignore
         service.onFirstPage = false;
 
         service.navigateBack([]);
 
-        expect(backSpy).to.have.been.calledOnceWithExactly();
+        expect(backSpy).toHaveBeenCalledTimes(1);
+        expect(backSpy).toHaveBeenCalledWith();
     });
```
Comment (julian-christl) [inline src/test/javascript/spec/service/navigation-util.service.spec.ts:45]:
```suggestion
    });
    
```

File: src/test/javascript/spec/service/navigation-util.service.spec.ts [hunk 3]
```diff
@@ -36,32 +29,37 @@ describe('Navigation Util Service', () => {
...
+        expect(router.navigate).toHaveBeenCalledWith(['a', 'b', 'c']);
     });
 
     it('should use without optional', () => {
         service.navigateBackWithOptional(['a', 'b', 'c'], undefined);
 
-        expect(router.navigateSpy).to.have.been.calledOnceWithExactly(['a', 'b', 'c']);
+        expect(router.navigate).toHaveBeenCalledTimes(1);
+        expect(router.navigate).toHaveBeenCalledWith(['a', 'b', 'c']);
     });
```
Comment (julian-christl) [inline src/test/javascript/spec/service/navigation-util.service.spec.ts:58]:
```suggestion
    });
    
```

**Review by julian-christl — CHANGES_REQUESTED** [review]

File: src/test/javascript/spec/component/admin/audits.component.spec.ts [hunk 4]
```diff
@@ -33,155 +34,152 @@ function getDate(isToday = true) {
...
+                mockActivatedRoute.setParameters({
+                    sort: 'id,desc',
+                });
             });
+    });
+
+    describe('today function', () => {
+        it('should set toDate to current date', () => {
+            comp.ngOnInit();
+            expect(comp.toDate).toBe(getDate());
```
[the remark this answers is not in the document]
Comment (MDK5) [reply src/test/javascript/spec/component/admin/audits.component.spec.ts:75]: Probably. If the tests happen to be executed exactly at 0:00 and the onInit gets the old date, but the expecrt the new, the tests will probably fail. I think this is out of scope for this PR, though.

**Review by julian-christl — APPROVED** [review]
Thanks for the changes.

**Review by lschlesinger — APPROVED** [review]
Looks good, let's merge :)

**Pull request merged**